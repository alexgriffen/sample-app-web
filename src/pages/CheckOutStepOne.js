import React from 'react';

class CheckOutStepOne extends React.Component {
	render() {
		return <h1>CheckOut: Your Information</h1>
	}
}

export default CheckOutStepOne
