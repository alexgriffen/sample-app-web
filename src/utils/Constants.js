export const FONT_FAMILY = 'Roboto, Arial, Helvetica, sans-serif';
