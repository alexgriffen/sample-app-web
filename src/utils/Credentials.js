import { VALID_PASSWORD,VALID_USERNAMES } from './Constants';

/**
 * Verify the credentials
 *
 * @param {string} username
 * @param {string} password
 *
 * @return {boolean}
 */
export function verifyCredentials(username, password) {
  if (password !== VALID_PASSWORD) {
    return false;
  }

  if (!VALID_USERNAMES.includes(username)) {
    return false;
  }

  // If we're here, we had a valid username and password.
  // Store the username in our session storage.
  sessionStorage.setItem('session-username', username);

  return true;
}

/**
 * Check if this is a problem user
 *
 * @return {boolean}
 */
export function isProblemUser() {
  return getSessionUsername() === 'problem_user';
}

/**
 * Check if this is a performance user
 *
 * @return {boolean}
 */
export function isPerformanceGlitchUser() {
  return getSessionUsername() === 'performance_glitch_user';
}

/**
 * Check if this a logged out user
 *
 * @return {boolean}
 */
export function isLockedOutUser() {
  return getSessionUsername() === 'locked_out_user';
}

/**
 * Check if the user is logged in with a valid username
 *
 * @return {boolean}
 */
export function isLoggedIn(){
  const sessionUsername = getSessionUsername();
  const isValidUsername = VALID_USERNAMES.includes(sessionUsername);

  return isValidUsername && sessionUsername !== 'locked_out_user';
}

/**
 * Get the session username
 *
 * @return {string}
 */
export function getSessionUsername() {
  return sessionStorage.getItem('session-username');

}
