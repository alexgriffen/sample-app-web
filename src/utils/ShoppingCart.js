import { CART_CONTENTS } from './Constants';

/**
 * Get the current content of the cart that is stored in the session storage
 *
 * @return {Array}
 */
export function getCartContent(){
  let curContents = sessionStorage.getItem(CART_CONTENTS);

  return curContents ? JSON.parse(curContents) : [];
}
