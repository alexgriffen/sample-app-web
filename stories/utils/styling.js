import { colors } from '../../src/utils/Colors';

export const style = {
  componentWrapper: {
    border: `1px solid ${colors.lightGray}`,
    margin: '20px auto',
    padding: 50,
    width: '75%',
  },
  grayComponentWrapper: {
    backgroundColor: colors.gray,
    border: `1px solid ${colors.lightGray}`,
    margin: '20px auto',
    padding: 50,
    width: '75%',
  },
  stateContainer: {
    fontFamily: 'Roboto, Helvetica, Arial, sans-serif',
    color: colors.gray,
    marginTop: 20,
  },
  stateWhiteContainer: {
    fontFamily: 'Roboto, Helvetica, Arial, sans-serif',
    color: colors.white,
    marginTop: 20,
  },
  lightText: {
    color: colors.lightGray,
  },
};
