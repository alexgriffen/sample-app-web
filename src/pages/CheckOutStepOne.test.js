import React from 'react';
import { shallow } from 'enzyme';
import CheckOutStepOne from './CheckOutStepOne'
import * as Credentials from '../utils/Credentials';

describe('CheckOutStepOne', () => {
  it('should render correctly for a logged in user', () => {
    const isLoggedInSpy = jest.spyOn(Credentials, 'isLoggedIn');
    isLoggedInSpy.mockReturnValue(true);

    const component = shallow(<CheckOutStepOne />);

    expect(isLoggedInSpy).toHaveBeenCalled();
    expect(component).toMatchSnapshot();

    isLoggedInSpy.mockClear();
  });

  it('should render correctly for a logged out user', () => {
    const isLoggedInSpy = jest.spyOn(Credentials, 'isLoggedIn');
    isLoggedInSpy.mockReturnValue(false);

    const component = shallow(<CheckOutStepOne />);

    expect(isLoggedInSpy).toHaveBeenCalled();
    expect(component).toMatchSnapshot();

    isLoggedInSpy.mockClear();
  });
});
