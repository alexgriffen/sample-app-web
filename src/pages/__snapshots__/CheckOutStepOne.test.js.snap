// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutStepOne should render correctly for a logged in user 1`] = `
<h1>
  CheckOut: Your Information
</h1>
`;

exports[`CheckOutStepOne should render correctly for a logged out user 1`] = `
<Redirect
  to="./"
/>
`;
